src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal partial class AbstractSuppressionCodeFixProvider
+    {
+        /// <summary>
+        /// Helper methods for pragma suppression add/remove batch fixers.
+        /// </summary>
+        private static class PragmaBatchFixHelpers
+        {
+            public static CodeAction CreateBatchPragmaFix(
+                AbstractSuppressionCodeFixProvider suppressionFixProvider,
+                Document document,
+                ImmutableArray<IPragmaBasedCodeAction> pragmaActions,
+                ImmutableArray<Diagnostic> pragmaDiagnostics,
+                FixAllContext fixAllContext)
+            {
+                // This is a temporary generated code action, which doesn't need telemetry, hence suppressing RS0005.
+#pragma warning disable RS0005 // Do not use generic CodeAction.Create to create CodeAction
```

Comment:
Current design is there is no difference between baseline all or suppress individual from error list. A future enhancement might add more state fields to SuppressMessage to allow differentiating the reason for suppression, workflow state (e.g. postponed till next milestone), etc.